src/vs/workbench/services/editor/common/editorGroupsService.ts

```diff
@@ -774,31 +774,18 @@ export interface IEditorGroup {
 	isActive(editor: EditorInput | IUntypedEditorInput): boolean;
 
 	/**
-	 * Selects the editor in the group. If active is set to true,
-	 * it will be the active editor in the group.
-	 */
-	selectEditor(editor: EditorInput, active?: boolean): Promise<void>;
-
-	/**
-	 * Selects the editors in the group. If activeEditor is provided,
-	 * it will be the active editor in the group.
-	 */
-	selectEditors(editors: EditorInput[], activeEditor?: EditorInput): Promise<void>;
-
-	/**
-	 * Unselects the editor in the group. If the editor is not specified, unselects the active editor.
-	 */
-	unSelectEditor(editor: EditorInput): Promise<void>;
-
-	/**
-	 * Unselects the editors in the group. If the editor is not specified, unselects the active editor.
+	 * Whether the editor is selected in the group.
 	 */
-	unSelectEditors(editors: EditorInput[]): Promise<void>;
+	isSelected(editor: EditorInput): boolean;
 
 	/**
-	 * Whether the editor is selected in the group.
+	 * Set a new selection for this group. This will replace the current
+	 * selection with the new selection.
+	 *
+	 * @param activeSelectedEditor the editor to set as active selected editor
+	 * @param inactiveSelectedEditors the inactive editors to set as selected
 	 */
-	isSelected(editor: EditorInput): boolean;
+	setSelection(activeSelectedEditor: EditorInput, inactiveSelectedEditors: EditorInput[]): void;
```

Comment:
I like this! Maybe `inactiveSelectedEditors` could be an optional parameter? 

Btw, this method needs to return `Promise<void>`.